lib/dao.js

```diff
@@ -509,6 +509,41 @@ DataAccessObject.upsert = function(data, options, cb) {
   if (id === undefined || id === null) {
     return this.create(data, options, cb);
   }
+  var doValidate = undefined;
+  if (options.validate === undefined) {
+    if (Model.settings.validateUpsert === undefined) {
+      if (Model.settings.automaticValidation !== undefined) {
+        doValidate = Model.settings.automaticValidation;
+      }
+    } else {
+      doValidate = Model.settings.validateUpsert;
+    }
+  } else {
+    doValidate = options.validate;
+  }
+
+  var forceId = this.settings.forceId;
+  if (forceId) {
+    options = Object.create(options);
+    options.validate = !!doValidate;
+    if (doValidate) {
+      Model.findById(id, options, function(err, model) {
+        if (err || !model) return cb(err || errorModelNotFound(id));
```

Comment:
I think is line is trying to be too clever. I am proposing to rewrite this bit to make it easier to understand:

```js
if (err) return cb(err);
if (!model) return cb(errorModelNotFound(id));
```